apps/settings/js/modules/dialog_manager.js

```diff
@@ -177,12 +177,21 @@ define(function(require) {
         return self._getPanel(dialog.panel.id);
       })
       .then(function(panel) {
-        // 4: call beforeShow
+        // 4: Inject cancel(), submit() functions for panel use
```

Comment:
ohoho, because this is clearer for me to break them into several steps, so if you prefer to make them together, I can update that.